src/server/mod.rs

```diff
@@ -833,6 +831,25 @@ where
                             let msg = format!("Restarting at {}. Now send STORE or RETRIEVE.", offset);
                             Ok(Reply::new(ReplyCode::FileActionPending, &*msg))
                         }
+                        Command::MDTM { file } => {
+                            ensure_authenticated!();
+                            let session = session.lock()?;
+                            let storage = Arc::clone(&session.storage);
+                            match storage.stat(&session.user, &file).wait() {
+                                Ok(meta) => match meta.modified() {
+                                    Ok(system_time) => {
+                                        let chrono_time: chrono::DateTime<chrono::offset::Utc> = system_time.into();
+                                        let formatted = chrono_time.format(RFC3659_TIME);
+                                        Ok(Reply::new(ReplyCode::FileStatus, formatted.to_string().as_str()))
```

Comment:
I feel like this code should have some test coverage.

Reply:
I feel so too I'm just not entirely sure how with the current state of the code.

Reply:
I'd like to refactor things a bit first to make it a bit more testable.